libnaucrates/src/statistics/CJoinStatsProcessor.cpp

```diff
@@ -294,33 +295,75 @@ CJoinStatsProcessor::SetResultingJoinStats
 		GPOS_ASSERT(NULL != outer_histogram);
 		GPOS_ASSERT(NULL != inner_histogram);
 		BOOL is_input_empty = CStatistics::IsEmptyJoin(outer_stats, inner_side_stats, IsLASJ);
-
 		CDouble local_scale_factor(1.0);
 		CHistogram *outer_histogram_after = NULL;
 		CHistogram *inner_histogram_after = NULL;
-		JoinHistograms
-				(
-						mp,
-						outer_histogram,
-						inner_histogram,
-						pred_info,
-						outer_stats->Rows(),
-						inner_side_stats->Rows(),
-						&outer_histogram_after,
-						&inner_histogram_after,
-						&local_scale_factor,
-						is_input_empty,
-						join_type,
-						DoIgnoreLASJHistComputation
-				);
 
-		output_is_empty = JoinStatsAreEmpty(outer_stats->IsEmpty(), output_is_empty, outer_histogram, inner_histogram, outer_histogram_after, join_type);
+		// When we have any form of equi join with join condition of type f(a)=b,
+		// we calculate the NDV of such a join as NDV(b) ( from Selinger et al.)
+		if (CStatsPred::EstatscmptEqNDVOuter == stats_cmp_type)
```

Comment:
I meant doing something like this:
```suggestion
		if (CStatsPred::EstatscmptEqNDVOuter == stats_cmp_type)
		{
		    inner_histogram = outer_histogram;
		} 
		else if (CStatsPred::EstatscmptEqNDVInner == stats_cmp_type)
		{
		    outer_histogram = inner_histogram;
		}
		
		JoinHistograms	
				(	
						mp,	
						outer_histogram,	
						inner_histogram,	
						pred_info,	
						outer_stats->Rows(),	
						inner_side_stats->Rows(),	
						&outer_histogram_after,	
						&inner_histogram_after,	
						&local_scale_factor,	
						is_input_empty,	
						join_type,	
						DoIgnoreLASJHistComputation	
				);
```

Reply:
Implemented thanks